integration/use-optionals-deprecated-only/test.ts

```diff
@@ -0,0 +1,640 @@
+/* eslint-disable */
+import * as _m0 from "protobufjs/minimal";
+import Long = require("long");
+
+export const protobufPackage = "optionalstest";
+
+/** @deprecated */
+export enum StateEnum {
+  UNKNOWN = 0,
+  ON = 1,
+  OFF = 2,
+  UNRECOGNIZED = -1,
+}
+
+export function stateEnumFromJSON(object: any): StateEnum {
+  switch (object) {
+    case 0:
+    case "UNKNOWN":
+      return StateEnum.UNKNOWN;
+    case 1:
+    case "ON":
+      return StateEnum.ON;
+    case 2:
+    case "OFF":
+      return StateEnum.OFF;
+    case -1:
+    case "UNRECOGNIZED":
+    default:
+      return StateEnum.UNRECOGNIZED;
+  }
+}
+
+export function stateEnumToJSON(object: StateEnum): string {
+  switch (object) {
+    case StateEnum.UNKNOWN:
+      return "UNKNOWN";
+    case StateEnum.ON:
+      return "ON";
+    case StateEnum.OFF:
+      return "OFF";
+    case StateEnum.UNRECOGNIZED:
+    default:
+      return "UNRECOGNIZED";
+  }
+}
+
+export interface OptionalsTest {
+  id: number;
+  /** @deprecated */
+  state?: StateEnum | undefined;
+  long: number;
+  /** @deprecated */
+  truth?:
+    | boolean
+    | undefined;
+  /** @deprecated */
+  description?:
+    | string
+    | undefined;
+  /** @deprecated */
+  data?: Uint8Array | undefined;
+  repId: number[];
+  /** @deprecated */
+  repState: StateEnum[];
```

Comment:
Should this be `repState?` since it is deprecated? Maybe I am not understanding the purpose of the flag.

Reply:
It looks like there is a check to say that repeated fields should not be marked as optional 
... @tufandevrim was there a reason to not include this for repeateds?

Reply:
@paralin @stephenh so sorry for the delay. My original plan was keeping repeated fields out of `deprecatedOnly` config but it really makes sense to make the repeated fields also optional if they are deprecated. I just updated the code, please review it when you get a chance. Thanks again for the review and feedback.